src/code-considerations/criteria-for-inclusion.md

```diff
@@ -0,0 +1,70 @@
+# Criteria for inclusion in `std`
+
+**Note**: *The process for deciding what does and does not get added to the standard
+library is too complex to reduce to a set of concrete rules. Ultimately we rely
+upon the expertise of members of the library API team, but here we attempt to
+document the reasoning that has been used historically to guide contributors
+and team members and to ensure long term consistency in how these decisions are
+made even as the team membership naturally changes over time.*
+
+## Reasons For Inclusion
+
+* APIs where there is a clear best implementation
+* APIs where there is a strong benefit to having only one, such as common
+  traits (e.g. `Error`, the `Fail` trait from `failure` crate showed the cost
+  of competing error traits and how it causes ecosystem fractures)
+* Core value propositions: APIs that we want to use in examples for our
+  learning materials to demonstrate rust's core value propositions. (e.g.
+  scoped threads simplifying examples demonstrating thread and memory safety)
+* Unsafe primitives: APIs that reduce the need of or centralize difficult to
+  implement unsafe code
+* Portability: APIs that are not dependent upon any platform's abstractions or
+  features
+
+## Reasons Against Inclusion
+
+* APIs where there are multiple legitimate alternatives with different tradeoffs
+* APIs that are excessively complex or difficult to maintain (e.g. Needle api
+  or `rand` crate)
+* API instability
+    * All new APIs will be subject to `std`'s perma 1.0 stability policy, so
+      certain classes of APIs that are more likely to require breaking changes
+      such as cryptography generally face a higher bar
+
+## Reasons that may seem like criteria but aren't
+
+* APIs that need access to the host operating system
+* APIs that are in other languages' standard libraries
+* APIs in crates with high download numbers
```

Comment:
That's a criteria for an unstable API, but not a criteria for a stable one.

Reply:
I would say it's more about public-ness than stability? Like an API which is used by the compiler but not otherwise a good fit for std should be a dep of the compiler, not in std. An API which is used internally by std, but not a good fit externally should be a private API. Stability shouldn't really come into it.

Reply:
@nrc Sure, either interpretation works. But either way those aren't public APIs of the standard library.